horace_core/sqw/@SQWDnDBase/IX_dataset_1d.m

```diff
@@ -0,0 +1,52 @@
+function wout = IX_dataset_1d (w)
+% Convert 1D sqw object into IX_dataset_1d
+%
+%   >> wout = IX_dataset_1d (w)
+
+% Original author: T.G.Perring
+%
+% $Revision:: 1759 ($Date:: 2020-02-10 16:06:00 +0000 (Mon, 10 Feb 2020) $)
+
+
+% Check input
+if isempty(w)
+    error('sqw object is an empty array')
+end
+for i=1:numel(w)
+    if dimensions(w(i))~=1
+        if numel(w)==1
+            error('sqw object is not one dimensional')
+        else
+            error('Not all elements in the array of sqw objects are one dimensional')
+        end
+    end
+end
+
+% Fill output
+wout=IX_dataset_1d;
+if numel(w)>1, wout(numel(w))=wout; end  % allocate array
+
+for i=1:numel(w)
```

Comment:
references to w(i).data should be updated to w(i).data_ to ensure the functions work with DnD objects